smarts/sstudio/genhistories.py

```diff
@@ -301,10 +318,19 @@ def _cal_heading(self, window):
             return self._prev_heading
         vhat = (n - c) / s
         r = math.atan2(vhat[1], vhat[0])
-        self._prev_heading = (r - math.pi / 2) % (2 * math.pi)
+        new_heading = (r - math.pi / 2) % (2 * math.pi)
+        if self._max_angular_velocity:
+            # XXX: could try to divide by sim_time delta here instead of assuming .1s
+            angular_velocity = (new_heading - self._prev_heading) / 0.1
+            if abs(angular_velocity) > self._max_angular_velocity:
+                new_heading = (
+                    self._prev_heading
+                    + np.sign(angular_velocity) * self._max_angular_velocity * 0.1
+                )
+        self._prev_heading = new_heading
```

Comment:
Maybe factor this out to a method to avoid code duplication.

Reply:
Good point, will do!